src/ripple/protocol/PublicKey.h

```diff
@@ -58,12 +58,14 @@ namespace ripple {
 */
 class PublicKey
 {
-protected:
-    std::size_t size_ = 0;
-    std::uint8_t buf_[33]; // should be large enough
+private:
+    bool empty_{true};
+    std::array<std::uint8_t, 33> buf_; // all supported public keys are 33 bytes
```

Comment:
although the empty is just used to save from needlessly copying data, the iterator functions would stop working because begin wouldn't be accessible, so nevermind :)

Reply:
I like the idea of using `optional`, but yeah, it complicates the iterators. Let's keep the bool member for now.